app/src/main/java/org/eyeseetea/malariacare/data/database/iomodules/dhis/importer/PullController.java

```diff
@@ -178,4 +197,13 @@ public void cancel() {
     public boolean isPullActive() {
         return PULL_IS_ACTIVE;
     }
+
+    public class LocalAsyncPull extends AsyncTask<Void, Void, Void> {
```

Comment:
@idelcano PullLocalDataSource receive a callback as parameter then better that inside this class execute in background and not from PullController equal to PullRemoteDataSource